src/PaddedViews.jl

```diff
@@ -2,16 +2,20 @@ __precompile__(true)
 
 module PaddedViews
 using Base: OneTo, tail
+using OffsetArrays
 
 export PaddedView, paddedviews
 
 """
     datapadded = PaddedView(fillvalue, data, sz)
-    datapadded = PaddedView(fillvalue, data, indices)
+    datapadded = PaddedView(fillvalue, data, padded_indices)
+    datapadded = PaddedView(fillvalue, data, padded_indices, data_indices)
 
 Create a padded version of the array `data`, where any elements within
-the span of `indices` not assigned in `data` will have value
-`fillvalue`.
+the span of `padded_indices` not assigned in `data` will have value
+`fillvalue`. If a second set of indices `data_indices` is not supplied
+it is assumed the array `data` spans the indices from the first up until
+`size(data)`, otherwise `data` spans the specified `data_indices`.
```

Comment:
This doesn't explain what the version with an `NTuple{N,Int}` for `data_indices` does. (That's described implicitly for the third argument with the `sz` line, though one could argue that should be more explicit too.)